packages/cras.rb

```diff
@@ -0,0 +1,64 @@
+require 'package'
+
+class Cras < Package
+  description 'ChromeOS Audio Server'
+  homepage 'https://www.chromium.org/chromium-os/chromiumos-design-docs/cras-chromeos-audio-server'
+  version 'stabilize-12249.B'
+  source_url 'https://chromium.googlesource.com/chromiumos/third_party/adhd/+/refs/heads/stabilize-12249.B/cras/README?format=TEXT'
+  source_sha256 'b65b959f4ad842a219a637413b8702676378a0b5d18b5e413c78c211b3fb133c'
+
+  depends_on 'alsa_lib'
+  depends_on 'ladspa'
+  depends_on 'iniparser'
+  depends_on 'speexdsp'
+  depends_on 'sbc'
+  depends_on 'dbus'
+
+  def self.build
+    system 'git', 'clone', 'https://chromium.googlesource.com/chromiumos/third_party/adhd', '-b', version, '.'
+    Dir.chdir('cras') do
+      system 'cat << _EOF_ > 10-cras.conf
+pcm.cras {
+    type cras
+    hint {
+        show on
+        description "Chromium OS Audio Server"
+    }
+}
+ctl.cras {
+    type cras
+}
+# Default: route all audio through the CRAS plugin.
+pcm.!default {
+    type cras
+    hint {
+        show on
+        description "Default ALSA Output (currently Chromium OS Audio Server)"
+    }
+}
+ctl.!default {
+    type cras
+}
+_EOF_'
+      system './git_prepare.sh'
+      system './configure',
+             "--prefix=#{CREW_PREFIX}",
+             "--libdir=#{CREW_LIB_PREFIX}",
+             '--disable-static',
+             '--disable-webrtc-apm',
+             '--enable-sse42',
+             '--enable-avx',
+             '--enable-avx2',
+             '--enable-fma'
+      system 'make'
+    end
+  end
+
+  def self.install
+    Dir.chdir('cras') do
+      system 'make', "DESTDIR=#{CREW_DEST_DIR}", "install"
+      FileUtils.mkdir_p CREW_DEST_PREFIX + '/share/alsa/alsa.conf.d/'
+      FileUtils.install '10-cras.conf', CREW_DEST_PREFIX + '/share/alsa/alsa.conf.d/'
+    end
+  end
```

Comment:
Missing `def self.postinstall` message to explain how to start and configure the server?

Reply:
No need. The server built in to Chrome OS is already running. All you need to do is run an application that uses ALSA.

Reply:
Apparently, you don't have much experience running daemons.  Any config change will require a server restart to pick up the change.

Reply:
No, it won't. The reading of configuration is a runtime setting.

Reply:
Where is this documented?  I have a difficult time believing that is possible.

Reply:
Under the "How" section of the homepage:

> Put an audio server in between Chromium and ALSA that can change how audio flows *on the fly*.

Emphasis added.

Reply:
Also, this is ALSA configuration, not CRAS configuration. ALSA has no server, nor does it have a daemon, so a server reset wouldn't even be possible.

Reply:
Well, technically it is a cras config since you are passing alsa settings to the server.  But this is very interesting.  I'll try to see if this actually works.